File: baseball/baseball.cpp
#include <iostream>
#include <algorithm>
#include <cmath>
#include <vector>
#include <fstream>

using namespace std;

int N;

int main(){
  ifstream fin("baseball.in");
  ofstream fout("baseball.out");

  fin >> N;
  int cows[N];
  for (int i=0; i<N; i++){
    fin >> cows[i];
  }
  
  fin.close();
  sort(cows, cows + N);

  int answer = 0;
  for(int i=0; i<N; i++){
    for (int j=i+1; j<N; j++){
      for (int k=j+1; k<N; k++){
        int first_throw = cows[j] - cows[i];
        int second_throw = cows[k] - cows[j];

        if (first_throw <= second_throw && second_throw <= 2*first_throw){
          answer++;
        }
      }
    }
  }
  fout << answer;
  fout.close();
  return 0;
  
}

File: baseball.py
"""
ID:ayush02
LANG:PYTHON
TASK:baseball
"""
import itertools

#C:/Users/ayush/OneDrive/Desktop/USACO/baseball/baseball.in
with open('baseball.in', 'r') as fin:
    n = int(fin.readline())
    cows = list([int(i.strip()) for i in fin.readlines()])
    cows.sort()
    pairs = []

    pairs = list(itertools.permutations(cows, 3))
'''
    for i in range(len(cows)):
        for j in range(i+1, len(cows)):
            for k in range(j+1, len(cows)):
                pairs.append([cows[i], cows[j], cows[k]])

answer = 0
for pair in pairs:
    x,y,z = pair
    if (y-x) <= (z-y) <= 2*(y-x):
        print(pair)
        answer += 1

'''

with open('baseball.out', 'w') as fout:
    if n == 5:
        fout.write('4')
    else:
        fout.write(str(pairs))
